diff --git a/hosts/broccoli/dhcpcd.nix b/hosts/broccoli/dhcpcd.nix
index 0b5b4ee..2566d13 100644
--- a/hosts/broccoli/dhcpcd.nix
+++ b/hosts/broccoli/dhcpcd.nix
@@ -4,7 +4,7 @@
   networking.dhcpcd = {
     enable = true;
     persistent = true;
-    allowInterfaces = [ "eno1" ];
+    allowInterfaces = [ config.networking.interfaces.enp0s20f0.name ];
     extraConfig = ''
       # Disable ipv6 router solicitation
       noipv6rs
diff --git a/hosts/broccoli/dhcpd4.nix b/hosts/broccoli/dhcpd4.nix
index ce06327..e4439ad 100644
--- a/hosts/broccoli/dhcpd4.nix
+++ b/hosts/broccoli/dhcpd4.nix
@@ -5,7 +5,7 @@
   ];
   services.dhcpd4 = {
     enable = true;
-    interfaces = [ "trusted" "iot" "guest" "mgmt" ];
+    interfaces = with config.networking.interfaces; [ trusted.name iot.name guest.name mgmt.name ];
     machines = [
       {
         ipAddress = "192.168.88.201";
diff --git a/hosts/broccoli/hardware-configuration.nix b/hosts/broccoli/hardware-configuration.nix
index 1e533ea..9972cce 100644
--- a/hosts/broccoli/hardware-configuration.nix
+++ b/hosts/broccoli/hardware-configuration.nix
@@ -5,26 +5,22 @@
 
 {
   imports =
-    [
-      (modulesPath + "/installer/scan/not-detected.nix")
+    [ (modulesPath + "/installer/scan/not-detected.nix")
     ];
 
-  boot.initrd.availableKernelModules = [ "xhci_pci" "ehci_pci" "ahci" "usb_storage" "usbhid" "sd_mod" ];
+  boot.initrd.availableKernelModules = [ "ehci_pci" "ahci" "xhci_pci" "usbhid" "usb_storage" "sd_mod" ];
   boot.initrd.kernelModules = [ ];
   boot.kernelModules = [ "kvm-intel" ];
   boot.extraModulePackages = [ ];
 
   fileSystems."/" =
-    {
-      device = "/dev/disk/by-uuid/7180e5f6-388b-4a94-acb7-bf79e15fe5e8";
+    { device = "/dev/disk/by-uuid/c4dc8beb-d97e-43e9-8168-a6aa55190985";
       fsType = "ext4";
     };
 
-  swapDevices = [ ];
-  zramSwap = {
-    enable = true;
-    swapDevices = 1;
-  };
+  swapDevices =
+    [ { device = "/dev/disk/by-uuid/6acebf9f-5838-4ff2-8fd2-c7ef319f25fe"; }
+    ];
 
   hardware.cpu.intel.updateMicrocode = lib.mkDefault config.hardware.enableRedistributableFirmware;
 }
diff --git a/hosts/broccoli/networking.nix b/hosts/broccoli/networking.nix
index 46c2eea..8d0da9e 100644
--- a/hosts/broccoli/networking.nix
+++ b/hosts/broccoli/networking.nix
@@ -15,24 +15,24 @@ in
     defaultGateway6.address = "2001:470:c:9::1";
     defaultGateway6.interface = "hurricane";
     sits.hurricane = {
-      dev = "eno1";
+      dev = "enp0s20f0";
       ttl = 255;
       remote = "66.220.18.42";
     };
     vlans = {
-      pubwan = { id = 10; interface = "enp1s0"; };
-      publan = { id = 20; interface = "enp1s0"; };
-      trusted = { id = 30; interface = "enp1s0"; };
-      iot = { id = 40; interface = "enp1s0"; };
-      guest = { id = 50; interface = "enp1s0"; };
-      mgmt = { id = 88; interface = "enp1s0"; };
+      pubwan = { id = 10; interface = "enp4s0"; };
+      publan = { id = 20; interface = "enp4s0"; };
+      trusted = { id = 30; interface = "enp4s0"; };
+      iot = { id = 40; interface = "enp4s0"; };
+      guest = { id = 50; interface = "enp4s0"; };
+      mgmt = { id = 88; interface = "enp4s0"; };
     };
     interfaces = {
-      eno1.useDHCP = true;
+      enp0s20f0.useDHCP = true;
       hurricane = {
         ipv6.addresses = [{ address = "2001:470:c:9::2"; prefixLength = 64; }];
       };
-      enp1s0 = {
+      enp4s0 = {
         ipv4.addresses = lib.mkForce [ ];
         ipv6.addresses = lib.mkForce [ ];
       };
diff --git a/hosts/broccoli/nftables.nix b/hosts/broccoli/nftables.nix
index 6c3287d..b41176d 100644
--- a/hosts/broccoli/nftables.nix
+++ b/hosts/broccoli/nftables.nix
@@ -7,7 +7,7 @@ in
   networking.nftables = {
     enable = true;
     ruleset = with config.networking.interfaces; ''
-      define DEV_WAN = ${eno1.name}
+      define DEV_WAN = ${enp0s20f0.name}
       define DEV_WAN6 = ${hurricane.name}
       define DEV_PUBWAN = ${pubwan.name}
       define DEV_PUBLAN = ${publan.name}
