diff --git a/src/partition/repart.c b/src/partition/repart.c
index 95cae94a8e..e9604551e8 100644
--- a/src/partition/repart.c
+++ b/src/partition/repart.c
@@ -7020,15 +7020,9 @@ static int parse_argv(int argc, char *argv[]) {
                  * former, if it is mounted, so that we have deterministic behaviour on systems where /usr/
                  * is vendor-supplied but the root fs formatted on first boot. */
                 r = path_is_mount_point("/sysusr/usr", NULL, 0);
-                if (r <= 0) {
-                        if (r < 0 && r != -ENOENT)
-                                log_debug_errno(r, "Unable to determine whether /sysusr/usr is a mount point, assuming it is not: %m");
-
-                        arg_root = strdup("/sysroot");
-                } else
+                if (r > 0) {
                         arg_root = strdup("/sysusr");
-                if (!arg_root)
-                        return log_oom();
+                }
         }
 
         arg_node = argc > optind ? argv[optind] : NULL;
