diff --git a/cmds/core/elvish/elvish.go b/cmds/core/elvish/elvish.go
index 91d6176e..9625c1bf 100644
--- a/cmds/core/elvish/elvish.go
+++ b/cmds/core/elvish/elvish.go
@@ -23,6 +23,7 @@ func main() {
 }
 
 func run(stdin, stdout, stderr *os.File, args []string) int {
+	args = append(args, "-rc=/etc/rc.elv")
 	return prog.Run([3]*os.File{stdin, stdout, stderr}, args, prog.Composite(buildinfo.Program, daemonStub{}, shell.Program{ActivateDaemon: nil}))
 }
 
