diff --git a/cmds/core/more/more.go b/cmds/core/more/more.go
index bb0f2ea3..38ffc7cd 100644
--- a/cmds/core/more/more.go
+++ b/cmds/core/more/more.go
@@ -33,20 +33,27 @@ var lines = flag.Int("lines", 40, "screen size in number of lines")
 
 func main() {
 	flag.Parse()
-	if flag.NArg() != 1 {
-		log.Fatal("more can only take one file")
-	}
+
 	if *lines <= 0 {
 		log.Fatal("lines must be positive")
 	}
 
-	f, err := os.Open(flag.Args()[0])
-	if err != nil {
-		log.Fatal(err)
+	var file *os.File
+
+	if stat, err := os.Stdin.Stat(); err == nil && stat.Mode()&os.ModeCharDevice == 0 {
+		file = os.Stdin
+	} else if flag.NArg() == 1 {
+		f, err := os.Open(flag.Args()[0])
+		if err != nil {
+			log.Fatal(err)
+		}
+		file = f
+		defer file.Close()
+	} else {
+		log.Fatal("bad usage")
 	}
-	defer f.Close()
 
-	scanner := bufio.NewScanner(f)
+	scanner := bufio.NewScanner(file)
 	for i := 0; scanner.Scan(); i++ {
 		if (i+1)%*lines == 0 {
 			fmt.Print(scanner.Text())
