diff --git a/src/soc/mediatek/mt8183/memlayout.ld b/src/soc/mediatek/mt8183/memlayout.ld
index 0acd174c..0d265e37 100644
--- a/src/soc/mediatek/mt8183/memlayout.ld
+++ b/src/soc/mediatek/mt8183/memlayout.ld
@@ -44,7 +44,7 @@ SECTIONS
 	DRAM_START(0x40000000)
 	DRAM_DMA(0x40000000, 1M)
 	POSTRAM_CBFS_CACHE(0x40100000, 1M)
-	RAMSTAGE(0x40200000, 256K)
+	RAMSTAGE(0x40200000, 2M)
 
 	BL31(0x54600000, 0x60000)
 }
diff --git a/src/soc/mediatek/mt8192/include/soc/memlayout.ld b/src/soc/mediatek/mt8192/include/soc/memlayout.ld
index 150bfdde..04b25256 100644
--- a/src/soc/mediatek/mt8192/include/soc/memlayout.ld
+++ b/src/soc/mediatek/mt8192/include/soc/memlayout.ld
@@ -54,7 +54,7 @@ SECTIONS
 	DRAM_START(0x40000000)
 	DRAM_DMA(0x40000000, 1M)
 	POSTRAM_CBFS_CACHE(0x40100000, 2M)
-	RAMSTAGE(0x40300000, 256K)
+	RAMSTAGE(0x40300000, 2M)
 
 	BL31(0x54600000, 0x60000)
 }
