diff --git a/src/libcamera/pipeline/simple/simple.cpp b/src/libcamera/pipeline/simple/simple.cpp
index 35c29cec..e8a0b9f4 100644
--- a/src/libcamera/pipeline/simple/simple.cpp
+++ b/src/libcamera/pipeline/simple/simple.cpp
@@ -265,6 +265,7 @@ static const SimplePipelineInfo supportedDevices[] = {
 	{ "mxc-isi", {}, false },
 	{ "qcom-camss", {}, true },
 	{ "sun6i-csi", {}, false },
+	{ "rockchip-cif", {}, true },
 };
 
 bool isRaw(const StreamConfiguration &cfg)
